File: laravel-magic-english-api/app/Models/Vocabulary.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Vocabulary extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'notebook_id',
        'word',
        'ipa',
        'meaning',
        'part_of_speech',
        'example',
        'cefr_level',
        'review_count',
        'last_reviewed_at',
    ];

    protected $casts = [
        'last_reviewed_at' => 'datetime',
    ];

    /* ================= Relations ================= */

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function notebook()
    {
        return $this->belongsTo(Notebook::class);
    }

    public function reviews()
    {
        return $this->hasMany(VocabularyReview::class);
    }
}

File: lib/project/dto/vocabulary.dart
class Vocabulary {
  final int id;
  final String word;
  final String meaning;
  final String partOfSpeech;
  final String ipa;
  final String example;
  final String cefrLevel;

  const Vocabulary({
    required this.id,
    required this.word,
    required this.meaning,
    required this.partOfSpeech,
    required this.ipa,
    required this.example,
    required this.cefrLevel,
  });

  static Vocabulary fromApiJson(Map<String, dynamic> json) {
    final rawId = json['id'];
    final id = rawId is int ? rawId : int.tryParse(rawId?.toString() ?? '') ?? 0;

    final word = (json['word'] ?? json['name'] ?? '').toString();
    final meaning = (json['meaning'] ?? '').toString();
    final partOfSpeech = (json['part_of_speech'] ?? json['type'] ?? '').toString();
    final ipa = (json['ipa'] ?? json['phonetic'] ?? '').toString();
    final example = (json['example'] ?? '').toString();
    final cefrLevel = (json['cefr_level'] ?? json['level'] ?? '').toString();

    return Vocabulary(
      id: id,
      word: word,
      meaning: meaning,
      partOfSpeech: partOfSpeech,
      ipa: ipa,
      example: example,
      cefrLevel: cefrLevel,
    );
  }
}
